src/js/components/DataTable/Header.js

```diff
@@ -91,8 +128,21 @@ const Header = ({
             verticalAlign,
             size,
           }) => {
-            let content =
-              typeof header === 'string' ? <Text>{header}</Text> : header;
+            // don't apply background or border on other contents of the cell
+            const contentStyles = { ...theme.dataTable.header };
+            delete contentStyles.background;
+            delete contentStyles.border;
+
+            let content;
+            if (typeof header === 'string') {
+              content = <Text {...contentStyles.font}>{header}</Text>;
```

Comment:
Shouldn't `color` should be passed here?

Reply:
If color is defined, then it would we placed on the box surrounding this and propagate down to the Text

Reply:
For clarity, I've included `color` in the `textProps` section of the utility function added in a recent commit.